onnxruntime_extensions/tools/pre_post_processing/steps/nlp.py

```diff
@@ -0,0 +1,449 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import onnx
+from collections import OrderedDict
+
+from typing import List, Optional, Tuple, Union
+from ..step import Step
+
+
+class SentencePieceTokenizer(Step):
+    def __init__(self, tokenizer, domain: str = "ai.onnx.contrib", name: Optional[str] = None):
```

Comment:
Do we need to take a hard dependency at this level on a specific tokenizer or could you just pass in the required info (bos_token_id and vocab_file) so this code is more generic?

Ideally the Step implementations have minimal external dependencies.

Reply:
Make sence. Will pass a Dict which contains all infos.

Reply:
A Dict is a little opaque. Could we maybe define a dataclass with the 3 fields so it's obvious what's required?

Reply:
Done. Add a new class for passing  tokenizer parameters